runtime/Cpp/runtime/src/Recognizer.h

```diff
@@ -11,7 +11,7 @@ namespace antlr4 {
 
   class ANTLR4CPP_PUBLIC Recognizer {
   public:
-    static const size_t EOF = std::numeric_limits<size_t>::max();
+    static const size_t EOF = (size_t)-1; // std::numeric_limits<size_t>::max(); doesn't work in VS 2013.
```

Comment:
Neither type of cast will be have any runtime overhead. The reason I prefer static_cast over a C-style cast is that the compiler enforces that the cast is safe. That way, when examining code, you only need to look for occurrences of reinterpret_cast to check for potential unsafe casting. This is also a rule on one of Scott Meyer's "Effective C++" books.